src/results.rs

```diff
@@ -0,0 +1,33 @@
+use std::fmt;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct ParsingError {
+    desc: String,
```

Comment:
I guess it could be switched to something like:

```rust
enum ParsingError {
  Other(String)
}
```

So the intent to use an enum is clear for the future of the API